automapper/custom_types.py

```diff
@@ -0,0 +1,10 @@
+from typing import Any, Callable, Dict, Iterable, Optional, Type, TypeVar
+
+from automapper.path_mapper import MapPath
+
+# Custom Types
+S = TypeVar("S")
+T = TypeVar("T")
+ClassifierFunction = Callable[[Type[T]], bool]
+SpecFunction = Callable[[Type[T]], Iterable[str]]
+FieldsMap = Optional[Dict[str | MapPath, Any]]
```

Comment:
I have enabled Github Actions to run checks on your branch and it seems to be failing.
Maybe makes sense for now to put these back as these are only used in 1 file. Not very confident in these `TypeVar` types that I've created a while ago. Need to refresh my memory.